packages/service-library/src/servicelib/long_running_tasks/_task.py

```diff
@@ -187,9 +187,21 @@ def _get_tracked_task(
                 if with_task_context and (
                     tasks[task_id].task_context != with_task_context
                 ):
+                    logger.debug(
+                        "Raising TaskNotFoundError for task_id=%s because %s != %s",
+                        task_id,
+                        f"{tasks[task_id].task_context=}",
+                        f"{with_task_context}=",
+                        exc_info=True,
```

Comment:
I think this `exc_info` will have no effect here since you are raising the exception **after** you are logging.

Reply:
yep, I agree. After our discussion yesterday I will remove these log statements and only correct the stack info thing we discussed in other log statements I made.